### ISSUE TITLE
Suggestion: reuse connection when syncing

### ISSUE THREAD

**Issue by dae** [ankidroid/Anki-Android#4568]
Hi guys,

I've just noticed that AnkiDroid is creating a new TCP/IP connection for each request it makes, instead of sending multiple requests over the existing connection. This means that the expensive SSL handshake is actually happening for every single request, which adds a lot of latency (close to 2 secs per request here!). If my Googling is correct, you should be able to have it reuse the existing connection by holding on to a copy of ThreadSafeClientConnManager for the duration of the sync, and reusing it instead of creating a new one for each request.

**Comment by mikehardy** [comment]
Here's the relevant code where it is created fresh:
https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/libanki/sync/HttpSyncer.java#L229

And the use of req() is in a context where there is information about the part of the sync being performed, so there should be enough information to implement this pretty quickly:
https://github.com/ankidroid/Anki-Android/blob/master/AnkiDroid/src/main/java/com/ichi2/libanki/sync/RemoteServer.java#L58

**Comment by david-allison-1** [comment]
Can be seen with the following. grep logcat for CONNECTION:

Sometimes flaky, may need a few app restarts and being run under debugger to appear in logcat when added in `AnkiDroidApp`.

```java
        Logger frameLogger = Logger.getLogger(Http2.class.getName());
        frameLogger.setLevel(Level.FINE);
        ConsoleHandler handler = new ConsoleHandler();
        handler.setLevel(Level.FINE);
        handler.setFormatter(new SimpleFormatter() {
            @Override public String format(LogRecord record) {
                return Util.format("%s%n", record.getMessage());
            }
        });
        frameLogger.addHandler(handler);
```

**Comment by mikehardy** [comment]
released as 2.10alpha70 as soon as google finishes processing it
